lang/c++/impl/Node.cc

```diff
@@ -79,7 +89,30 @@ void Name::check() const
 
 bool Name::operator == (const Name& n) const
 {
-    return ns_ == n.ns_ && simpleName_ == n.simpleName_;
+	if (ns_ == n.ns_ && simpleName_ == n.simpleName_)
+		return true;
+
+	const std::string& myFullName = fullname();
+	const std::string& theirFullName = n.fullname();
+
+	for (const auto& theirAlias : n.aliases_)
```

Comment:
We have not used `auto` so far in our code base. I think it was because of some warning in some compilers that its meaning has changed or something like this. Is it possible to avoid using auto in these places before we figure out a consistent convention on use of `auto`?